.github/workflows/docs-ci.yml

```diff
@@ -396,3 +396,57 @@ jobs:
       - name: Upload JVM Error Logs
         uses: actions/upload-artifact@v4
         if: failure()
+
+  protodoc:
+    needs: [makedirs]
+    runs-on: ubuntu-22.04
+    concurrency:
+      group: protobufdoc-${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup JDK 11
+        id: setup-java-11
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Set JAVA_HOME
+        run: echo "JAVA_HOME=${{ steps.setup-java-11.outputs.path }}" >> $GITHUB_ENV
+
+      - name: Setup gradle properties
+        run: |
+          .github/scripts/gradle-properties.sh >> gradle.properties
+          cat gradle.properties
+
+      - name: Generate Protobuf Docs
+        run: ./gradlew generateProtobuf
+
+      - name: Get Deephaven Version
+        id: dhc-version
+        run: echo "version=$(cat build/version)" >> $GITHUB_OUTPUT
+
+      - name: Upload Protobuf Docs
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: protobuf-docs-${{ steps.dhc-version.outputs.version }}
+          path: 'proto/proto-backplane-grpc/build/generated/source/proto/main/proto-doc/single-html/index.html'
+
+      - name: Deploy Protobuf Docs
+        if: ${{ github.event_name == 'push' }}
+        uses: burnett01/rsync-deployments@5.2
+        with:
+          switches: -rlptDvz --delete
+          path: proto/proto-backplane-grpc/build/generated/source/proto/main/proto-doc/single-html/index.html
+          remote_path: deephaven-core-v2/${{ github.ref_name }}/client-api/protobuf/
```

Comment:
I think you'll need to update the `makedirs` job to make sure this directory already exists.